asdf_astropy/converters/table/table.py

```diff
@@ -0,0 +1,101 @@
+from asdf.extension import Converter
+from asdf.tags.core.ndarray import NDArrayType
+
+
+class ColumnConverter(Converter):
+    tags = ["tag:stsci.edu:asdf/core/column-*"]
+
+    types = [
+        "astropy.table.column.Column",
+        "astropy.table.column.MaskedColumn",
+    ]
+
+    def to_yaml_tree(self, obj, tag, ctx):
+        node = {
+            "data": obj.data,
+            "name": obj.name
+        }
+
+        if obj.description:
+            node["description"] = obj.description
+
+        if obj.unit:
+            node["unit"] = obj.unit
+
+        if obj.meta:
+            node["meta"] = obj.meta
+
+        return node
+
+    def from_yaml_tree(self, node, tag, ctx):
+        from astropy.table import Column, MaskedColumn
+        from numpy.ma.core import MaskedArray
+
+        data = node["data"]
+        if isinstance(data, NDArrayType):
+            # TODO: Why doesn't NDArrayType work?  This needs some research
+            # and documentation.
+            data = data._make_array()
+
+        if isinstance(data, MaskedArray):
```

Comment:
This check is missing in astropy and as a result the mask is lost on read.  Do we need to fix that bug in astropy too?